hoodie-spark/src/main/java/com/uber/hoodie/DataSourceUtils.java

```diff
@@ -142,4 +143,29 @@ public static HoodieRecord createHoodieRecord(GenericRecord gr, Comparable order
     HoodieRecordPayload payload = DataSourceUtils.createPayload(payloadClass, gr, orderingVal);
     return new HoodieRecord<>(hKey, payload);
   }
+
+  @SuppressWarnings("unchecked")
+  public static JavaRDD<HoodieRecord> dropDuplicates(JavaSparkContext jssc,
+                                                     JavaRDD<HoodieRecord> incomingHoodieRecords,
+                                                     Map<String, String> parameters) throws Exception {
+    HoodieReadClient<HoodieRecordPayload> client;
+    try {
+      HoodieWriteConfig writeConfig = HoodieWriteConfig
+          .newBuilder()
+          .withPath(parameters.get("path"))
+          .withProps(parameters).build();
+      client = new HoodieReadClient<>(jssc, writeConfig);
+      return client.tagLocation(incomingHoodieRecords.map(r -> new HoodieRecord<>(r.getKey(), r.getData())))
```

Comment:
These conversions back and forth are ugly and inefficient, but it is the only way I found to deal with type conversion and compiler check.

Wondering if there is a better way to do this?

Reply:
Here is an alternate implementation : 

  public static JavaRDD<HoodieRecord<HoodieRecordPayload>> dropDuplicates(JavaSparkContext jssc,
                                                     JavaRDD<HoodieRecord<HoodieRecordPayload>> incomingHoodieRecords,
                                                     Map<String, String> parameters) throws Exception {
    HoodieReadClient<HoodieRecordPayload> client;
    try {
      HoodieWriteConfig writeConfig = HoodieWriteConfig
          .newBuilder()
          .withPath(parameters.get("path"))
          .withProps(parameters).build();
      client = new HoodieReadClient<>(jssc, writeConfig);
      return client.tagLocation(incomingHoodieRecords).filter(r -> !r.isCurrentLocationKnown());
    } catch (DatasetNotFoundException e) {
      // this will be executed when there is no hoodie dataset yet
      // so no dups to drop
      return incomingHoodieRecords;
    }
  }